### ISSUE TITLE
Feature requests: Alphabetical sorting of artists & art movements ++

### ISSUE THREAD

**Issue by nightgolfer** [some9000/StylePile#53]
Absolutely love your plugin, fantastic work!

Would it be possible to make the various lists (artists, art movements, etc.) sort alphabetically? Both in the dropdowns and the galleries. See screenshot. 

Also, would it be possible to have more than X and Y sequential prompts?

Lastly, the placeholders ([X], [A], etc.) seem to be case sensitive ⨪ do they need to be?

I'm on version 85b549a1 btw.

![Screenshot 2023-04-14 21 03 46](https://user-images.githubusercontent.com/32328229/232134480-d80b1c8d-b91a-4749-b15b-cde7571ff9af.png)

Thanks!

**Comment by some9000** [comment]
Hi! I was just browsing by... Glad to hear you are enjoying it. Let's go point by point:

This is weird, since on my system artist names and everything else from dropdowns is A-Z sorted by default. I will look if there is some way that between systems it can read the files differently. Must be due to the way you run it in RunDiffusion.

As for more prompts - The system already digs into X, then digs into Y... Any deeper seems like a bit of an overkill. But, in case you hadn't noticed you can do multiple lines there. So you can do a couple of prompts with different elements  in the X field that can contain an [Y] and then have several lines in the Y field. That really should be enough.

Case sensitivity is something was planned to fix. Just ran into a little bug here and commented that part out.

I am currently working on a new heavily updated version with many fixed/improved features. If you don't mind being a bit of a beta tester you can grab it here. It is mostly functional just with some rough edges: https://github.com/some9000/StylePile/tree/Error-hunting

**Comment by nightgolfer** [comment]
Thanks for the response!

Sounds like it could be a RD thing with the sorting if it isn't happening on your setup.

Cool with case sensitivity, it's not like it was a huge thing, just something I noticed =)

I tried the link you sent here and in my other issue thread, but the link is 404. I looked through the branches and found 2.1 beta, but installing it by URL seems to fail, and because of the way the RD system is set up, I don't have direct access to the Extensions folder so can't do a manual install. Automatic install seems to just give me the current stable version.

Regarding X / Y / Z, I'm already using multiple lines in X and Y, which works really well. The reason I wanted Z was for a scenario like this:

Character Type X, Wearing Y, at Location Z.

With current setup I have to:

Character Type X, Wearing Y, at Random Location A.

Specifically, I was trying to make human characters with the head of Animal X, Wearing Y, at Location Z (but had to do Random Location A instead).

**Comment by some9000** [comment]
Yeah, I got a bit of a sync problem with GitHub and had to rename the latest testing version to something else. But yeah, in that different environment it's just not possible to manually change, I guess.

Either way, they next version's "official" release is not far off, so you should be about to try it out soon. I just need to regenerate a mountain of preview images and do some small fixes.

As for the terms. I guess it really depends on how many extra versions you have. In theory all the fields can hold a lot of lines. And in 2.1 you will be able to save all those X,Y and A-H values as presets. So then it will be worth it to do a bunch of typing (unless it's like 100 lines...). I will think about it. But, as mentioned, 3 level loops sound like a fairly niche case and may be even more confusing for users.

**Comment by nightgolfer** [comment]
No worries about the looping, if it's not a good idea I can find a manageable workaround 😊

If you spot anything that could be triggering the weird sorting on RunDiffusion, do let me know. I will get in touch with the team there and ask them if they can think of a reason this might be happening with their implementation.

Looking forward to 2.1! ⭐️

And one tiny, dumb question... will SP 2.1 work with SD 1.5? 😅

**Comment by some9000** [comment]
Of course it will. It just generates text, after all. 2.1 was just what I was looking at to work out result adjustments.
  
But the system per se works with all the custom models and so on. So no worries.
  

  

  

  
  
  
  
  
>

**Comment by nightgolfer** [comment]
Thank you 😊 Very new to all of this and wasn't completely sure 😅
